Protocol.md

```diff
@@ -300,6 +317,173 @@ The recipient:
 
   These script pub key forms include only standard forms accepted by the wider set of deployed Bitcoin clients in the network, which increase the chances of successful propagation to miners.
 
+# Channel
+
+## Overview
+
+The lifetime of a channel is composed of its establishment, and the possibility for each party to offer the settlement or renewal of the contract within the channel, as well as to close the channel.
+
+The establishment protocol is depicted in the following [state machine](#state-machine-representation).
+
+<img src="./images/dlc_channel_establish_ta.svg">
+
+The protocol for renewal, settlement and closing of the channel is shown in the following [state machine](#state-machine-representation).
+
+<img src="./images/dlc_channel_settle_renew.svg">
+
+### Revocation
+
+Revocation is done following the mechanism introduced in [this paper](https://eprint.iacr.org/2020/476).
+
+### State machine representation
+ The state machine is represented as a timed automaton augmented with:
+* channel synchronization representing message passing between the processes (`?` indicating the receiving of a message while `!` indicates its sending),
+* edge guards only enabling state transition to be taken when the condition is satisfied (e.g. `Invalid` being true),
+* location invariants requiring the associated condition to hold for the automaton to stay in the location,
+* updates, setting the value of free variables and clocks. 
+
+Note that transitions setting the variable `Invalid` to true are omitted for clarity.
+
+## The `offer_channel` message
+
+1. type: 43000 (`offer_channel_v0`)
+2. data:
+   * [`u32`: `protocol_version`]
+   * [`byte`:`contract_flags`]
+   * [`chain_hash`:`chain_hash`]
+   * [`32*byte`:`temporary_contract_id`]
+   * [`32*byte`:`temporary_channel_id`]
+   * [`contract_info`:`contract_info`]
+   * [`point`:`funding_pubkey`]
+   * [`point`:`revocation_base_point`]
+   * [`point`:`publish_base_point`]
+   * [`point`:`own_base_point`]
+   * [`point`:`first_per_update_point`]
+   * [`spk`:`payout_spk`]
+   * [`u64`:`payout_serial_id`]
+   * [`u64`:`offer_collateral_satoshis`]
+   * [`bigsize`:`num_funding_inputs`]
+   * [`num_funding_inputs*funding_input`:`funding_inputs`]
+   * [`spk`:`change_spk`]
+   * [`u64`:`change_serial_id`]
+   * [`u64`:`fund_output_serial_id`]
+   * [`u64`:`feerate_per_vb`]
```

Comment:
Do you intend to take the same `feerate_per_vb` for all transactions (`FundTx`, `RefundTx` and `CloseTx`)?

Reply:
That's what I do at the moment yes. Do you have a different suggestion? Also my thinking is to maybe have something to update the fee [like in LN](https://github.com/lightning/bolts/blob/master/02-peer-protocol.md#updating-fees-update_fee).

Reply:
We are doing the same thing and additionally we send a new `feerate_per_vb` whenever we extend/rollover a contract.

For our usecase in particular an `update_fee` message would not be needed as either you contact the other party to _rollover/extend_ your contract and provide a new `feerate_per_vb` or you closed the position with `CloseOffer` which also sending over a `feerate_per_vb`.

Reply:
Very good point. Though I wonder if we should have something to indicate that we want to accept a renewal but if a different fee rate?